pkg/processor/metricfrequencyprocessor/metriccache.go

```diff
@@ -45,13 +43,13 @@ func (mc *metricCache) Register(name string, dataPoint pdata.NumberDataPoint) {
 
 	internalCache, exists := mc.internalCaches[name]
 	if !exists {
-		newCache := newCache()
+		newCache := mc.newCache()
 		mc.internalCaches[name] = newCache
 		internalCache = newCache
 	}
 
 	key := dataPoint.Timestamp().String()
-	value := &DataPoint{Timestamp: dataPoint.Timestamp(), Value: dataPoint.Value()}
+	value := &DataPoint{Timestamp: dataPoint.Timestamp(), Value: dataPoint.DoubleVal()}
```

Comment:
we should support `IntVal` as well

Reply:
Support added with `getVal` function and NaN filtering.